src/sentry/web/frontend/project_settings.py

```diff
@@ -187,6 +200,24 @@ def clean_slug(self):
                                         'using that slug' % other.name)
         return slug
 
+    def clean_token(self):
+        token = self.cleaned_data.get('token')
+        if not token:
+            return
+        token_re = r'^[a-zA-Z0-9-]{1,40}$'
```

Comment:
There is _some_ limit. How long are you thinking? The value can't be say, 1MB long. :) I picked 40 as an arbitrary value since our auto generated value is a uuid which is 32 characters.

Just wondering what case you'd want to shove a JWT here.